### ISSUE TITLE
Switch as X integration disables zwave_js configuration

### ISSUE THREAD

**Issue by markaggar** [home-assistant/frontend#12641]
### The problem

I 'converted' nearly all of my Zwave switches to either lights or fans using the 'Switch as X' integration.  However, doing so means that I cannot write a new automation via the UI that triggers on state change of the device (e.g. it being turned on or off, or a zwave value changing).  Also once the switch is converted, you cannot view the zwave configuration for it anymore.

e.g. config of switch before - note the zwave config options

![image](https://user-images.githubusercontent.com/25288127/167230353-13eb68e2-6ee8-43f3-a269-0601ec9589f3.png)

after:  zwave info not available

![image](https://user-images.githubusercontent.com/25288127/167230401-fe92bae9-0975-4f7f-86f5-5e75b4989f34.png)

Trigger not available when switch as light

![image](https://user-images.githubusercontent.com/25288127/167230464-2eedc928-c207-4a0b-bbd5-fe97a2162096.png)

### What version of Home Assistant Core has the issue?

core-2022.5.1

### What type of installation are you running?

Home Assistant OS

### Integration causing the issue

Switch as X

### Link to integration documentation on our website

https://www.home-assistant.io/integrations/switch_as_x

**Comment by emontnemery** [comment]
@markaggar If you still have the problem where the device triggers don't work, please open another issue on core 👍

**Comment by markaggar** [comment]
Sorry for the late comment.  In 2022.5.4, I noticed that about half of my 50 switches had zwave JS as the first integration, with switch as x as the second integration.  As such I could configure the zwave settings for those devices.  However the other half had switch as x as the first integration and couldn't be configured for zwave.  Installed 2022.5.5 and it's the same.  Seems totally random as to which device gets which integration listed first.  Not sure if the fix that was checked in addresses this or when it will show up in a mainline release.
